### ISSUE TITLE
Getting the same object twice from a `createState` does not return objects that are equal

### ISSUE THREAD

**Issue by rjclaasen** [avkonst/hookstate#291]
Hello, thanks again for all of your hard work! It's been simply amazing to work with.

I have a question/issue I'm curious to get a 2nd opinion on:

For testing some helper functions I would like to create a state outside of a component & pass it to those functions to verify they are working correctly with values contained in a state. (they use `.value` internally so I can't just pass objects)

To my understanding, `useState` can not be used outside of components, but `createState` can, so that is the method I should be using for this.

However, I am running into an issue with object equality not working as expected. It seems that when you pull the same object twice from the same `createState` state, they are not equal. If you pull the same object from `useState` twice, they are the equal, so there seems to be some difference in functionality I am not understanding.

Example inside component:

```jsx

const Component = () => {
  const created = createState({hello: ["hi"]}); // array for simplicity, but can be object too
  const used = useState({hello: ["hi"]});

  console.log(created.hello == created.hello); // false
  console.log(created.hello.value == created.hello.value); // false

  console.log(used.hello == used.hello); // true
  console.log(used.hello.value == used.hello.value); // true

  return </>;
}
```

For now this is only an issue in specs that do not touch React at all, as components use `useState` & pass its result to the helper functions and behave as expected. The issue only occurs when `useState` is not used before passing the state to the helper functions. 

Did I understand correctly that using `createState` is the recommended way to use a hookstate outside of components & that that would be a good way to mock a component passing a state to a helper function, or should I write some mock-state myself for use in testing?

Please let me know if I can help in any way & if you need any more information, I'd love to help out!

**Comment by avkonst** [comment]
> Did I understand correctly that using createState is the recommended way to use a hookstate outside of components
yes
> that that would be a good way to mock a component passing a state to a helper function
yes, I do it all the time
> or should I write some mock-state myself for use in testing?"
You can but I would recommend not to

Regarding the issue of references being different each time, it is done to avoid memory leaks, but I will have a look if it can be solved in Hookstate-4 where the new engine is capable to handle stable state references.

**Comment by avkonst** [comment]
implemented in 4.0.0-rc8
